types/tx/types.go

```diff
@@ -6,12 +6,13 @@ import (
 	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/tendermint/tendermint/crypto"
 )
 
 // MaxGasWanted defines the max gas allowed.
 const MaxGasWanted = uint64((1 << 63) - 1)
 
-var _, _ codectypes.UnpackInterfacesMessage = &Tx{}, &TxBody{}
+var _, _, _, _ codectypes.UnpackInterfacesMessage = &Tx{}, &TxBody{}, &AuthInfo{}, &SignerInfo{}
```

Comment:
I guess you want to check the interface implementations here.

1. I'm not sure if the first parameters will do any type assertion, example:

        var a, b IT = A{}, B{}

    I'm not sure that this will assert that `a` is of type `IT`.

2. Please add a comment that you are doing type checks:

        // Interface implementation checks
        var ( .... )

Reply:
Ok for 2.

1 works as expected, I tried adding a typo in the method, it doesn't compile.